src/pveclib/vec_int64_ppc.h

```diff
@@ -2175,6 +2180,97 @@ vec_cmpud_any_ne (vui64_t a, vui64_t b)
   return (result);
 }
 
+/** \brief Vector Load with gather unsigned doubleword from offsets.
+ *
+ *  For each doubleword element [i] of vra, load the doubleword
+ *  element at *array+vra[i]. Merge those doubleword elements and
+ *  return the resulting vector.
+ *
+ *  \note As effective address calculation is modulo 64-bits, signed or
+ *  unsigned doubleword offsets are equivalent.
+ *
+ *  |processor|Latency|Throughput|
+ *  |--------:|:-----:|:---------|
+ *  |power8   |   12  | 1/cycle  |
+ *  |power9   |   11  | 1/cycle  |
+ *
+ *  @param array Pointer to array of unsigned doublewords.
+ *  @param vra Vector of doubleword offsets.
+ *  @return vector doubleword containing elements loaded from
+ *  *array+vra[0] and *array+vra[1].
+ */
+static inline
+vui64_t
+vec_lvgudo (unsigned long long int *array, vui64_t vra)
+{
+  vui64_t rese0, rese1;
+
+  rese0 = vec_vlxsdx (vra[VEC_DW_H], array);
+  rese1 = vec_vlxsdx (vra[VEC_DW_L], array);
+  return  vec_permdi (rese0, rese1, 0);
+}
+
+/** \brief Vector Load with gather unsigned doubleword scaled indexed.
+ *
+ *  For each doubleword element [i] of vra, load the doubleword
+ *  element array[vra[i] * (1 << scale)]. Merge those doubleword
+ *  elements and return the resulting vector. Indexes are converted to
+ *  offsets from *array by shifting each doubleword left (3+scale) bits.
+ *
+ *  \note As effective address calculation is modulo 64-bits, signed or
+ *  unsigned doubleword indexes are equivalent.
```

Comment:
Yes but what is the type of the index/offset? which is separate from the type of the pointer. This is fact of modulo addition. 

This what the ISA says to the effect "we don't need two instructions here, one will do".

So I need to say more?

Reply:
Sorry, I guess my reference to `array` was incorrect, because the indexes are in `vra`. However, `vra` is of type `vui64_t`, which is `unsigned` as well. So, the caller has to deal with that type regardless.  Are you just saying that they could really be passing in a `vi64_t`? Since they have to deal with that type, I'm not sure how the statement helps...whatever they pass will be treated as `unsigned`.  Are you just saying "don't worry about it"?

Reply:
VRA could be vui64_t or vi64_t offsets and as address arithmetic is modulo 64-bit, it does not matter which. But as PVECLIB does not support type generic operations. I had to pick one. The user can use a cast to match the API.

For doubleword int, vui64_t is my default, unless the operation is explicitly signed (vec_srad vs vec_srd).

The questions is, which would be most consistent? Looking at the ABI (intrinsic Guide) for vec_ld, vec_lde, vec_st, the RA operand is **signed long long**. This implies that vec_vlxsdx and vec_vstxsdx should use **signed long long** as well. Etc, etc up through vec_stvsudo vec_lvsudo, ...

I am not opposed to this. but this pull request is getting messy. change this now or clean this up as is and submit a separate pull request to change vra|ra to signed?

Reply:
> VRA could be vui64_t or vi64_t offsets and as address arithmetic is modulo 64-bit, it does not matter which. But as PVECLIB does not support type generic operations. I had to pick one. The user can use a cast to match the API.

This is my point. Regardless of what the user picks, it will be treated as `unsigned` by the API. Indeed the user has to cast it as such if they use `signed`.  So, the comment is unnecessary, and maybe confusing. "Pick signed or unsigned as long as it is unsigned."

> The questions is, which would be most consistent?

Agree with your point that in _Power Vector Intrinsic Programming Reference_ and the _Power ISA_, effective address calculations use signed offsets. In the absence of a reason to use unsigned offsets here, they should probably be converted to signed.  How and when to do that is up to you.

Reply:
It would be nice to have actual usage example (like vector sin/cos with table look up). I suspect those are unsigned. 

What I propose is commit this (int64) with ra unsigned for now. Them move on to vector double (f64) using signed ra and see what this means for the unit test code. Then can see if it matters.